src/chunk_index.c

```diff
@@ -925,6 +925,55 @@ chunk_index_mark_clustered(Oid chunkrelid, Oid indexrelid)
 	heap_close(rel, AccessShareLock);
 }
 
+/*
+ * Create versions of every index over src_chunkrelid over chunkrelid.
+ * Returns the relids of the new indexes created.
+ * New indexes are in the same order as RelationGetIndexList.
+ */
+List *
+chunk_index_duplicate(Oid src_chunkrelid, Oid dest_chunkrelid)
```

Comment:
Can we refactor this and chunk_index_clone to use common code?

Reply:
This is surprisingly difficult because the two functions have very different control flow. `chunk_index_duplicate` starts with a chunk uses that to get the hypertable and from that the indices, while `chunk_index_clone` goes index -> chunk -> hypertable.